src/test/java/com/nvs/store/service/ProductServiceTest.java

```diff
@@ -0,0 +1,76 @@
+package com.nvs.store.service;
+
+import com.nvs.store.models.product.Product;
+import com.nvs.store.repository.ProductRepository;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import java.math.BigDecimal;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
```

Comment:
SpringBootTest using many configurations that are not required to test ProductService. Instead of spring boot test here can be used mockito framework.

Reply:
done